### PR TITLE
Refactor User Portal Post & Comment Components with ProfileAvatarDisplay 

### LINKED ISSUE #5155: Phase 2.1: Refactor User Portal Post & Comment Components with ProfileAvatarDisplay
## Overview
Integrate ProfileAvatarDisplay into high‑visibility User Portal components (posts, comments, start post modal).

Plan: https://github.com/PalisadoesFoundation/talawa-admin/issues/5130#issuecomment-3672637070

## Files to Modify
1) src/components/UserPortal/PostCard/PostCard.tsx (creator + commenters)
2) src/components/UserPortal/CommentCard/CommentCard.tsx
3) src/components/UserPortal/StartPostModal/StartPostModal.tsx

## Acceptance Criteria
- All avatars replaced with ProfileAvatarDisplay
- Layout/styling preserved
- Alt/i18n strings used
- Unit tests updated and passing
- No TS/ESLint issues

## Before/After (PostCard excerpt)
```tsx
// Before
{creator.avatarURL ? <img src={creator.avatarURL} /> : <Avatar name={name} />}

// After
<ProfileAvatarDisplay imageUrl={creator.avatarURL} fallbackName={name} alt={t('userAvatar',{name})} size={40} radius={20} enableModal />
```

### PR DESCRIPTION
**Pull request by khushalbhasin4488**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #5155

**Snapshots/Videos:**

https://github.com/user-attachments/assets/252d95bf-f438-4f6d-8b81-c4fd48962e86

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [X] I have written tests for all new changes/features
- [X] I have verified that test coverage meets or exceeds 95%
- [X] I have run the test suite locally and all tests pass

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

* **Improvements**
  * Avatar display refined across posts, comments, and create-post: standardized sizes, optional enlarge-to-modal (keyboard-accessible), shows user name in modal when available, and improved border behavior.

* **Style**
  * Normalized “more options” wording to lowercase in English, Spanish, and French; Chinese wording clarified for better readability.

* **Tests**
  * Updated tests to match new avatar accessibility text and modal behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by palisadoes** [comment]
FYI

This is a tracking page for the major refactoring issues.

- https://github.com/PalisadoesFoundation/talawa-admin/issues/4997

**Review by palisadoes — CHANGES_REQUESTED** [review]
Exclude this file from the PR. It is out of scope:

```
.github/workflows/scripts/css_check.py

```

File: src/components/UserPortal/StartPostModal/StartPostModal.module.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.divider {
+  margin: 0;
+}
```
Comment (Jashan32) [inline src/components/UserPortal/StartPostModal/StartPostModal.module.css]: don't add this file in ```startPostModal``` as its content is moved to ```src/shared-components/posts/createPostModal```

Comment (palisadoes) [reply src/components/UserPortal/StartPostModal/StartPostModal.module.css]: @khushalbhasin4488 Please fix this
@Jashan32 is there a way to lint for this and similar changes?

Comment (Jashan32) [reply src/components/UserPortal/StartPostModal/StartPostModal.module.css]: @palisadoes
The directories are change too frequently for path based linting to be reliable.

**Comment by palisadoes** [comment]
Please fix the conflicting files

**Comment by khushalbhasin4488** [comment]
@palisadoes  I have added video in the description of this pr .

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all resolved items and include the following points:

1. **Test Coverage & Quality Validation:**
    1. Validate whether any updated test files:
       1. Cover all use cases including edge cases, error scenarios, and success paths
       2. Have valid, properly structured tests following Jest and React Testing Library best practices
       3. Provide 100% test code coverage for all modified/added functions, components, and branches
       4. Include proper mocking of GraphQL queries/mutations, external dependencies, and API calls
       5. Test both user interactions and state changes comprehensively
       6. Require any additional tests for uncovered scenarios
    1. For each modified file with logic or UI, verify corresponding test files exist and are updated:
        1. Check that test files follow naming conventions (_.test.tsx, _.test.ts, _.spec.tsx, _.spec.ts)
        1. Validate that tests don't include disabled tests (it.skip, describe.skip) unless explicitly documented
        1. Ensure mock data cleanup is performed (as required by pre-commit hooks)

1. **Code Quality & Repository Standards:**
    1. Verify TypeScript compliance:
         1. No use of `any` type without explicit justification
         2. Proper interface/type definitions for all props, state, and function parameters
         3. No TypeScript errors or warnings
    1. Validate GraphQL implementation:
         1. Queries/mutations are properly typed and follow existing patterns in src/GraphQl/
         2. No duplicate or redundant queries (check for pagination inconsistencies)
         3. Error handling is implemented for all GraphQL operations
    1. Check React component best practices:
         1. Functional components with proper hooks usage
         2. Material-UI component patterns are followed consistently
         3. No prop drilling; proper use of context when needed
         4. Proper cleanup in useEffect hooks
    1. Verify internationalization (i18n):
         1. All user-facing strings use i18next translation keys
         2. New translation keys are added to locale files
         3. No hardcoded English text in UI components
    1. Validate accessibility:
         1. Proper ARIA labels and roles
         2. Keyboard navigation support
         3. Screen reader compatibility
  
1. **Documentation & Code Comments:**
   1. Verify auto-generated documentation is updated (as enforced by Check-AutoDocs job)
   2. Check for adequate JSDoc comments on public functions and complex logic
   3. Ensure README.md or related docs are updated if new features are added
   4. Validate that Table of Contents is updated (as required by pre-commit hooks)

1. **Identify Issues:**
   1. Lines of code that are missing tests (list specific line numbers)
   2. Any unnecessary files that have been submitted (build artifacts, node_modules, .env files, IDE configs)
   3. Unused imports or dead code (as checked by knip)
   4. Console.log statements or debug code left in production files
   5. Files that should be excluded from the PR scope

1. **PR Requirements Validation:**
   1. Verify the PR meets the requirements of the linked issue:
      1. All acceptance criteria are addressed
      2. No out-of-scope changes are included
   2. Validate compliance with all previously suggested improvements and comments:
      1. All review comments have been addressed
      2. Requested changes have been implemented correctly
   3. Check PR metadata:
      1. Proper title and description
      2. Linked to appropriate issue(s)
      3. Target branch is correct (typically `develop`)
   4. Verify no sensitive information is exposed (as checked by Check-Sensitive-Files job)

1. **CI/CD Pipeline Compatibility:**
   1. Validate that code changes won't break any of the 17+ CI jobs
   2. Ensure pre-commit hook requirements are met:
      1. Code formatting (Prettier)
      2. Linting (ESLint)
      3. Type checking (TypeScript)
   3. Check that changes don't introduce performance regressions for the 12-shard test execution

1. **Security & Best Practices:**
   1. No security vulnerabilities introduced
   2. No exposed API keys, tokens, or credentials
   3. Proper input validation and sanitization
   4. No SQL injection or XSS vulnerabilities in any data handling

1. **Implementation Correctness:**
   1. For the specific implementation file(s) being tested:
      1. Verify the implementation logic is correct and free of bugs
      1. Validate that queries target the correct database tables/entities
      1. Check that foreign key relationships are correctly enforced
      1. Confirm business logic constraints are properly implemented
      1. Any bugs, logic errors, or incorrect implementations discovered in the file(s) under test MUST be fixed in this PR before approval

**Comment by palisadoes** [comment]
@coderabbitai why wasn't the missing test file discovered in via the .coderabbit.yaml file found at https://github.com/PalisadoesFoundation/talawa-admin/blob/develop/.coderabbit.yaml

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Please fix
    - https://github.com/PalisadoesFoundation/talawa-admin/pull/5431#issuecomment-3695056329
    - https://github.com/PalisadoesFoundation/talawa-admin/pull/5431#issuecomment-3695065158

1. Also this needs to be adjusted
        image

File: src/components/UserPortal/StartPostModal/StartPostModal.module.css (hunk 1 above)
Comment (palisadoes) [reply src/components/UserPortal/StartPostModal/StartPostModal.module.css]: @khushalbhasin4488 

1. Please address this comment.
2. Also, this needs to be fixed.

    image

**Comment by palisadoes** [comment]
@coderabbitai what should we add to the .coderabbit.yaml to ensure prompts are followed for automatic reviews when PRs are submitted or updated?

**Comment by khushalbhasin4488** [comment]
@palisadoes , here is the fixed ui 
Screenshot 2025-12-29 at 11 55 02 AM

**Comment by palisadoes** [comment]
@coderabbitai full review

**Review by palisadoes — APPROVED** [review]
